src/main/java/org/junit/internal/MethodSorter.java

```diff
@@ -49,16 +60,49 @@ public int compare(Method m1, Method m2) {
      *      (non-)bug #7023180</a>
      */
     public static Method[] getDeclaredMethods(Class<?> clazz) {
-        Comparator<Method> comparator = getSorter(clazz.getAnnotation(FixMethodOrder.class));
+        FixMethodOrder order = clazz.getAnnotation(FixMethodOrder.class);
+        Comparator<Method> comparator = getSorter(order);
 
         Method[] methods = clazz.getDeclaredMethods();
         if (comparator != null) {
             Arrays.sort(methods, comparator);
         }
+        if(order != null && order.value() == MethodSorters.RANDOM) {
+            Seed s = clazz.getAnnotation(Seed.class);
+            long seed = s == null ? generateSeed() : s.value();
+            randomize(methods, clazz.getName(), seed);
+        }
 
         return methods;
     }
 
+    // Same seed generation algorithm used in JDK 6
+    private static long generateSeed() {
+        return ++seedUniquifier + System.nanoTime();
+    }
+
+    /**
+     * Puts array of methods in random order.
+     * A seed for the random number generator is selected
+     * using the same method as the no-argument 
+     * constructor in java.util.Random.
+     * This seed is then displayed on stdout.
+     * @param methods the array of methods
+     * @param className the name of this class
+     */
+    private static void randomize(Method[] methods, String className, 
+        long seed) {
+        System.out.printf("Class: %s, Random ordering seed = 0x%xL%n", 
+            className, seed);
+        Random r = new Random(seed);
+        for(int i=methods.length-1; i>0; i--) {
```

Comment:
BTW, you could instead do:

  `Collections.shuffle(Arrays.asList(methods), r);`